### DISCUSSION TITLE
issue: preview panel doesn't show full text when mixing Arabic and English letters

### DISCUSSION THREAD

**Discussion by asaah18** [KDE/ghostwriter#677] [General]
Mixing Arabic and English letters miss up preview panel as follow:

- Show the symbol `�`, if the number of characters is **even**.
- Cut off the last three characters, if the number of characters is **odd**. 

Most probably because of the encoding of Arabic letters. Adding **additional characters** at the end of the document solves the problem.

Examples:

| Editor panel  | Preview panel | number of characters | additional characters |
| -------------  | ---------------  | ----------------------- | ----------------------- |
|  `أهلا hello`    |  `أهلا h`            | even                           |  No                             |
| `أهلا hello    ` |  `أهلا hello`      | even                           | Yes                              |
|  `أهلا helo`    |  `أهلا`                | odd                           |  No                             |
| `أهلا helo    ` |  `أهلا hello`      | odd                           | Yes                              |

Operating System: Windows 10

**Comment by tiagoft** [comment]
Hello! In addition, I have found that the preview, as well as the export, has some issues when it finds diacritics in general. In Portuguese, it means that á, é, ó and so on cause some weird behavior. What happens is that the preview (as well as the export) starts "lagging". For example, if I write: ÁÁÁÁ, the reviewer renders ÁÁ (that is, as if each diacritic counted as two characters when it is rendered).

Operating system: Windows 10.

**Reply by tiagoft** [reply]
Interesting update: I recompiled it under Linux and in these conditions the problem does not appear.

**Comment by tiagoft** [comment]
Ok, I found a possible solution for this.

Ghostwriter uses third-party programs that rely on the system's encoding. Cmark expects UTF-8, but Windows 10 uses something else by default. So, you need to go to control panel / region / change system locale and enable UTF-8 encoding (it says it's Beta but seems to work fine). After that, restart the system and everything works fine.

More info: https://www.quora.com/Why-doesnt-Microsoft-use-UTF-8-on-Windows-10

Cheers!!
